compiler/src/dotty/tools/dotc/core/Signature.scala

```diff
@@ -87,23 +82,27 @@ case class Signature(paramsSig: List[ParamSig], resSig: TypeName) {
   /** The degree to which this signature matches `that`.
    *  If parameter signatures are consistent and result types names match (i.e. they are the same
    *  or one is a wildcard), the result is `FullMatch`.
-   *  If only the parameter signatures are consistent, the result is `ParamMatch` before erasure and
-   *  `NoMatch` otherwise.
+   *  If only the parameter signatures are consistent, the result is either
+   *  `MethodNotAMethodMatch` (if one side is a method signature and the other isn't),
+   *  or `ParamMatch`.
    *  If the parameters are inconsistent, the result is always `NoMatch`.
    */
   final def matchDegree(that: Signature)(implicit ctx: Context): MatchDegree =
-    if (consistentParams(that))
-      if (resSig == that.resSig || isWildcard(resSig) || isWildcard(that.resSig)) FullMatch
-      else if (!ctx.erasedTypes) ParamMatch
-      else NoMatch
-    else NoMatch
+    if consistentParams(that) then
+      if resSig == that.resSig || isWildcard(resSig) || isWildcard(that.resSig) then
+        FullMatch
+      else if (this == NotAMethod) != (that == NotAMethod) then
+        MethodNotAMethodMatch
+      else
+        ParamMatch
+    else
+      NoMatch
 
   /** Does this signature potentially clash with `that` ? */
   def clashes(that: Signature)(implicit ctx: Context): Boolean =
     matchDegree(that) == FullMatch
 
-  /** name.toString == "" or name.toString == "_" */
-  private def isWildcard(name: TypeName) = name.isEmpty || name == tpnme.WILDCARD
+  private def isWildcard(name: TypeName) = name == tpnme.WILDCARD
```

Comment:
Rename to `isWildcardOrEmpty`

Reply:
Actually that commit removed the `name.isEmpty` check, so the current name of that method should be accurate.